yarn-project/noir-contracts/src/contracts/token_contract/src/account_interface.nr

```diff
@@ -0,0 +1,52 @@
+
+use dep::std::option::Option;
+
+use dep::aztec::context::{
+    PrivateContext,
+    PublicContext,
+    Context,
+};
+
+struct AccountContract {
+  address: Field,
+}
+
+impl AccountContract {
+  fn at(address: Field) -> Self {
+    Self {
+      address,
+    }
+  }
+
+  // Will call is_valid(_public) on the account contract
+  // Will insert the nullifier into the context
+  // Will revert if the authorization fails
+  fn is_valid(
+    self: Self, 
+    context: Context,
```

Comment:
I'm anticipating users will get confused as to why they need to wrap something that's already called `context` in a `Context`.

Reply:
Also, I don't understand Noir's `if` statements well enough, but this might do two lots of function calls (executing both branches of the `if / else`), which will slow down proofs significantly (there's a lot of hashing to make a function call). Calling the exact function will use far fewer constraints.